src/sleep/sleep.c

```diff
@@ -102,6 +112,7 @@ static int write_efi_hibernate_location(const HibernationDevice *hibernation_dev
                         &v,
                         SD_JSON_BUILD_PAIR_UUID("uuid", uuid),
                         SD_JSON_BUILD_PAIR_UNSIGNED("offset", hibernation_device->offset),
+                        SD_JSON_BUILD_PAIR_CONDITION(!!part_designator, "autoSwap", SD_JSON_BUILD_BOOLEAN(streq_ptr(part_designator, "swap"))),
```

Comment:
hmm, the really interesting bit here is this comparison, which is hidden down here. Can we maybe factor this out into a helper? i.e. determine_auto_swap() or so, which does the block_device_get_originating() call and does the comparison and just returns error/0/1?